src/test/java/com/microsoft/azure/kusto/kafka/connect/sink/it/ITSetup.java

```
@@ -0,0 +1,83 @@
+                name += "/";
+            }
+            JarEntry entry = new JarEntry(name);
+            entry.setTime(source.lastModified());
+            target.putNextEntry(entry);
+            target.closeEntry();
+            for (File nestedFile : Objects.requireNonNull(source.listFiles())) {
+                add(nestedFile, target);
+            }
+        } else {
+            JarEntry entry = new JarEntry(name);
+            entry.setTime(source.lastModified());
+            target.putNextEntry(entry);
+            try (BufferedInputStream in = new BufferedInputStream(Files.newInputStream(source.toPath()))) {
+                byte[] buffer = new byte[1024];
+                while (true) {
+                    int count = in.read(buffer);
+                    if (count == -1) {
+                        break;
+                    }
+                    target.write(buffer, 0, count);
+                }
+                target.closeEntry();
+            }
+        }
+    }
+
+    static ITCoordinates getConnectorProperties() {
+        String testPrefix = "tmpKafkaSinkIT_";
+        String appId = getProperty("appId", "", false);
+        String appKey = getProperty("appKey", "", false);
+        String authority = getProperty("authority", "", false);
+        String cluster = getProperty("cluster", "", false);
+        String database = getProperty("database", "e2e", true);
+        String defaultTable = testPrefix + UUID.randomUUID().toString().replace('-', '_');
+        String table = getProperty("table", defaultTable, true);
+        return new ITCoordinates(appId, appKey, authority, cluster, database, table);
+    }
+
+    private static String getProperty(String attribute, String defaultValue, boolean sanitize) {
+        String value = System.getProperty(attribute);
+        if (value == null) {
+            value = System.getenv(attribute);
+        }
+        value = StringUtils.isEmpty(value) ? defaultValue : value;
```

shouldnt we throw instead of getting default empty value?

---

Right again , this was a generic method. The caller skips the test if say APP id or cluster URL is not defined. The default comes handy for things like default DB name etc.